ocaml/xenops/domain.ml

```diff
@@ -211,15 +213,26 @@ let make ~xc ~xs vm_info uuid =
 			(* create cpu and memory directory with read only perms *)
 			List.iter (fun dir ->
 				let ent = sprintf "%s/%s" dom_path dir in
-				t.Xst.mkdir ent;
-				t.Xst.setperms ent roperm
+				mkdirperms t ent roperm
 			) [ "cpu"; "memory" ];
+
+
+			let mksubdirs base dirs perms =
+				List.iter (fun dir ->
```

Comment:
But yes, I agree that it would still be readable without the intermediate `ent` variable.